frontend/src/components/CommentInput.js

```diff
@@ -0,0 +1,186 @@
+import { useCallback, useMemo, useState } from 'react'
+import isHotkey from 'is-hotkey'
+import { Editable, withReact, Slate } from 'slate-react'
+import { createEditor } from 'slate'
+import { withHistory } from 'slate-history'
+import { useAuth } from '../contexts/AuthContext'
+import { axiosPreset } from '../axiosConfig'
+import { Box, Button } from '@chakra-ui/react'
+
+import {
+    BlockButton,
+    Element,
+    Leaf,
+    MarkButton,
+    Toolbar,
+    toggleMark,
+} from './SlateComponents'
+
+const HOTKEYS = {
+    'mod+b': 'bold',
+    'mod+i': 'italic',
+    'mod+u': 'underline',
+}
+
+const cleanInput = (val) => {
+    const res = []
+    let l = 0
+    let r = val.length - 1
+    while (l <= r) {
+        if (val[l]['children'][0]['text'] == '') {
+            l += 1
+        }
+        if (val[r]['children'][0]['text'] == '') {
+            r += -1
+        }
+
+        if (
+            val[l]['children'][0]['text'] != '' &&
+            val[r]['children'][0]['text'] != ''
+        ) {
+            break
+        }
+    }
+    for (let i = l; i <= r; i++) {
+        res.push(val[i])
+    }
+    return res
+}
+
+const invalidInput = (val) => {
+    if (val.length === 0) {
```

Comment:
seems like a redundant check. if `val.length` is 0 then the for loop below shouldn't run